python/graphstorm/gconstruct/id_map.py

```diff
@@ -100,18 +108,20 @@ def map_id(self, ids):
         tuple of tensors : the tensor of new IDs, the location of the IDs in the input ID tensor.
         """
         for id_ in self._ids:
```

Comment:
this loop is to test whether the dtype of keys matches the input array?

Reply:
Yes.